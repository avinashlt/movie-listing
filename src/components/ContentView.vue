<template>
<div class="content-view-wrapper">
  <div v-if="!movieDetails.Title" class="select-movie-text"> Select a movie to view more details</div>
  <div v-else>
    <el-row :gutter="10">
    <el-col :xs="12" :sm="12" :md="12" :lg="12" :xl="12">
      <div class="grid-content" v-if="movieDetails.Title">
              <div class="details-wrapper">
                  <h1 class= "movie-title">{{movieDetails.Title}}</h1>
                  <p class="movie-genre">{{movieDetails.Genre}}</p>
                  <p class="movie-description">{{movieDetails.Plot}}</p>
                  <p class="movie-language"><b>Language :</b> {{movieDetails.Language}}</p>
                  <p class="movie-director"><b>Director :</b> {{movieDetails.Director}}</p>
                  <p class="movie-actors"><b>Actors :</b> {{movieDetails.Actors}}</p>
                  <p class="movie-duration"><b>Duration :</b> {{movieDetails.Runtime}}</p>
              </div>
      </div>
    </el-col>
    <el-col :xs="12" :sm="12" :md="12" :lg="12" :xl="12"><div class="grid-content">
          <div class= "image-wrapper">
              <el-image class="poster-image" fit="cover" v-if="movieDetails.Poster === 'N/A'" src="https://placeimg.com/300/408/any/sepia'">
              </el-image>
              <el-image class="poster-image" fit="cover" v-else="movieDetails.Poster" v-bind:src="movieDetails.Poster">
              </el-image>

          </div>
    </div></el-col>
  </el-row>
  </div>

</div>
</template>
<style scoped>
  .grid-content {
    min-height: 100%;
  }
  .image-wrapper .poster-image {
        margin: 0 auto;
        display: table;
  }
  .select-movie-text{
    text-align : center;
  }

</style>

<script>
export default {
    name: 'ContentView',
    props : ['movieDetails'],
    data() {
        return {
        };
    }
}
</script>