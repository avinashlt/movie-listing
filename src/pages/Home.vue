<template>
    <el-container>
    <div class="left-nav-bar">
        <el-aside style="width:100%">
          <ListView :items="movies" v-on:searchChange="getSearchResults" v-on:movieItemClick="itemClickEvent"></ListView>
        </el-aside>
    </div>

    <div class="main-desktop-content">
        <el-container>
            <el-main>
                <ContentView :movieDetails = "movieDetails"></ContentView>
            </el-main>
        </el-container>
    </div>
    <div class="main-mobile-content">
        <el-dialog title="Movie Details" :visible.sync="dialogVisible" width="90%">
          <ContentView :movieDetails = "movieDetails"></ContentView>
        </el-dialog>
    </div>
    
    </el-container>
</template>

<style scoped>
  .left-nav-bar{
    width:300px;
    border-right : 1px solid var(--border-color);
  }
  .main-mobile-content {
    display:none;
  }
  .main-desktop-content {
      display : block;
      width: 100%;
  }

  @media (min-width: 320px) and (max-width: 480px) {
    .left-nav-bar{
      width:100%;
    }
    .main-desktop-content {
      display : none;
    }
    .main-mobile-content {
      display:block;
    }
  }
  
</style>

<script>
import ListView from './../components/ListView'
import ContentView from './../components/ContentView'
import MovieList from './../services/movie.js'
  export default {
    components : {
        ListView,
        ContentView
    }, 
    data() {
      return {
        movies : [],
        movieDetails : {},
        dialogVisible : false,
      }
    },
    methods : {
      getSearchResults(value, page){
        MovieList.getMovie(value, page).then(res => {
          this.movies = res || {};
        });
      },

      itemClickEvent(imdbID) {
        if(screen.width < 480)
        this.dialogVisible = true;
        this.getMovieDetails(imdbID);
      },

      getMovieDetails(imdbID) {
        MovieList.getMovieDetails(imdbID).then(res => {
          this.movieDetails = res || {};
        })
      }


    }
  };
</script>