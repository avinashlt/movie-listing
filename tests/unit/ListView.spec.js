import { createLocalVue, shallowMount} from '@vue/test-utils'; 
import Home from '../../src/pages/Home.vue'
import ContentView from '../../src/components/ContentView.vue'
import ElementUI from 'element-ui'; 

const localVue = createLocalVue();
localVue.use(ElementUI);   
describe('ListView.vue', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallowMount(ContentView, {localVue});
  });

  it('should render list view', () => { 
    expect(wrapper.find('.list-wrapper')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(Home, {
      propsData: {}
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})