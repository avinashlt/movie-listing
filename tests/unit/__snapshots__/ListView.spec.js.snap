// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView.vue matches snapshot 1`] = `
<el-container>
  <div class="left-nav-bar">
    <el-aside style="width: 100%;">
      <listview-stub items=""></listview-stub>
    </el-aside>
  </div>
  <div class="main-desktop-content">
    <el-container>
      <el-main>
        <contentview-stub moviedetails="[object Object]"></contentview-stub>
      </el-main>
    </el-container>
  </div>
  <div class="main-mobile-content">
    <el-dialog title="Movie Details" width="90%">
      <contentview-stub moviedetails="[object Object]"></contentview-stub>
    </el-dialog>
  </div>
</el-container>
`;
